app/src/frontend/building/data-components/numeric-data-entry.tsx

```diff
@@ -24,7 +34,12 @@ const NumericDataEntry: React.FunctionComponent<any> = (props) => { // TODO: rem
                 min={props.min || 0}
                 disabled={props.mode === 'view' || props.disabled}
                 placeholder={props.placeholder}
-                onChange={props.onChange}
+                onChange={e => 
+                    props.onChange(
+                        props.slug,
+                        e.target.value === '' ? null : parseFloat(e.target.value)
+                    )
+                }
```

Comment:
Parse numbers already inside NumericDataEntry and pass only the resulting value higher up, rather than the whole event. This is to enable correctly detecting changes. Otherwise, if the higher level form component deals with raw string data, initially the field might have a numeric value but after an edit it becomes a string, so detecting edits in the UI is more difficult. The NumericDataEntry, and other types of input components, are specific to the value type they represent, so the parsing of that type is embedded inside the component